lib/reader/src/parser.rs

```diff
@@ -40,11 +40,32 @@ pub fn parse_test<'a>(text: &'a str) -> Result<TestFile<'a>> {
     let mut parser = Parser::new(text);
     // Gather the preamble comments as 'Function'.
     parser.gather_comments(AnyEntity::Function);
+
+    let commands = parser.parse_test_commands();
+    let mut isa_spec = parser.parse_isa_specs()?;
+
+    if let isaspec::IsaSpec::Some(ref mut isa_vec) = isa_spec {
+        if isa_vec.len() == 1 {
+            // Remove it from the isa_vec just while we parse functions, avoiding a .clone()
+            parser.target_isa = Some(isa_vec.remove(0));
```

Comment:
Honestly I did that because I wasn't sure how to make it borrow check correctly with the references. I've just been having another look, even explicitly dropping the parser earlier doesn't help:

`    if let isaspec::IsaSpec::Some(ref mut isa_vec) = isa_spec {
        if isa_vec.len() == 1 {
            parser.unique_isa = Some(&*isa_vec[0]);
        }
    }

    let preamble_comments = parser.take_comments();
    let functions = parser.parse_function_list()?;

    drop(parser);

    Ok(TestFile {
        commands: commands,
        isa_spec: isa_spec,
        preamble_comments: preamble_comments,
        functions: functions,
    })`

`error: `(isa_spec:isaspec::IsaSpec::Some).0` does not live long enough
  --> lib\reader\src\parser.rs:64:1
   |
47 |     if let isaspec::IsaSpec::Some(ref mut isa_vec) = isa_spec {
   |                                   --------------- borrow occurs here
...
64 | }
   | ^ `(isa_spec:isaspec::IsaSpec::Some).0` dropped here while still borrowed
   |
   = note: values in a scope are dropped in the opposite order they are created`